WeatherApp/Core/Managers/LocationManager.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  LocationManager.swift
+//  WeatherApp
+//
+//  Created by Mohamed Ali on 28/06/2024.
+//
+
+import Foundation
+import CoreLocation
+
+final class LocationManager: NSObject, CLLocationManagerDelegate, ObservableObject {
```

Comment:
Why NSObject?

Reply:
Because the CLLocationManagerDeleagate protocol inherits from NSObjectProtocol which contains methods, we didn't implement here so we instead inherit from NSObject that already has Implemented all these methods.

Reply:
@MohamedAdel7710 Never worked so deeply with CLLocationManager, this is new for me.
الله ينور 👏🏻